src/A2A.AspNetCore/A2AJsonRpcProcessor.cs

```diff
@@ -138,7 +138,17 @@ private static T DeserializeOrThrow<T>(JsonElement jsonParamValue) where T : cla
             parms = null;
         }
 
-        return parms ?? throw new A2AException("Invalid parameters", A2AErrorCode.InvalidParams);
+        switch (parms)
+        {
+            case null:
+                throw new A2AException("Invalid parameters", A2AErrorCode.InvalidParams);
+            case MessageSendParams messageSendParams when messageSendParams.Message.Parts.Count == 0:
+                throw new A2AException("Message parts cannot be empty", A2AErrorCode.InvalidParams);
+            case TaskQueryParams taskQueryParams when taskQueryParams.HistoryLength < 0:

```

Comment:
Should the historylength check be <= 0? e.g. why would anybody require historyLength = 0?

```suggestion
            case TaskQueryParams taskQueryParams when taskQueryParams.HistoryLength <= 0:

```